zip-0312.rst

```diff
@@ -1,7 +1,447 @@
+FROST ``verify_signature_share`` function, but with its inputs modified relative
+to the ``randomizer`` as following:
+
+- ``PK_i = PK_i + G.ScalarBaseMult(randomizer)``
+- ``group_public_key = group_public_key + G.ScalarBaseMult(randomizer)``
+
+
+
+
+
+Ciphersuites
+------------
+
+FROST(Jubjub, BLAKE2b-512)
+''''''''''''''''''''''''''
+
+This ciphersuite uses Jubjub for the Group and BLAKE2b-512 for the Hash function ``H``
+meant to produce signatures indistinguishable from RedJubjub Sapling Spend
+Authorization Signatures as specified in [#protocol-concretespendauthsig]_.
+
+- Group: Jubjub [#protocol-jubjub]_ with base point :math:`\mathcal{G}^{\mathsf{Sapling}}`
+  as defined in [#protocol-concretespendauthsig]_.
+
+  - Order: :math:`r_\mathbb{J}` as defined in [#protocol-jubjub]_.
+  - Identity: as defined in [#protocol-jubjub]_.
+  - RandomScalar(): Implemented by returning a uniformly random Scalar in the range
+    \[0, ``G.Order()`` - 1\]. Refer to {{frost-randomscalar}} for implementation guidance.
+  - SerializeElement(P): Implemented as :math:`\mathsf{repr}_\mathbb{J}(P)` as defined in [#protocol-jubjub]_
+  - DeserializeElement(P): Implemented as :math:`\mathsf{abst}_\mathbb{J}(P)` as defined in [#protocol-jubjub]_,
+    returning an error if :math:`\bot` is returned. Additionally, this function
+    validates that the resulting element is not the group identity element,
+    returning an error if the check fails.
+  - SerializeScalar: Implemented by outputting the little-endian 32-byte encoding
+    of the Scalar value.
+  - DeserializeScalar: Implemented by attempting to deserialize a Scalar from a
+    little-endian 32-byte string. This function can fail if the input does not
+    represent a Scalar in the range \[0, ``G.Order()`` - 1\].
+
+- Hash (``H``): BLAKE2b-512 [#BLAKE]_ (BLAKE2b with 512-bit output and 16-byte personalization string),
+  and Nh = 64.
+
+  - H1(m): Implemented by computing BLAKE2b-512("FROST_RedJubjubR", m), interpreting
+    the 64 bytes as a little-endian integer, and reducing the resulting integer
+    modulo ``G.Order()``.
+  - H2(m): Implemented by computing BLAKE2b-512("Zcash_RedJubjubH", m), interpreting
+    the 64 bytes as a little-endian integer, and reducing the resulting integer
+    modulo ``G.Order()``.
+    (This is equivalent to :math:`\mathsf{H}^\circledast(m)`, as defined by
+    the :math:`\mathsf{RedJubjub}` scheme instantiated in [#protocol-concretereddsa]_.)
+  - H3(m): Implemented by computing BLAKE2b-512("FROST_RedJubjubN", m), interpreting
+    the 64 bytes as a little-endian integer, and reducing the resulting integer
+    modulo ``G.Order()``.
+  - H4(m): Implemented by computing BLAKE2b-512("FROST_RedJubjubM", m).
+  - H5(m): Implemented by computing BLAKE2b-512("FROST_RedJubjubC", m).
+  - HR(m): Implemented by computing BLAKE2b-512("FROST_RedJubjubA", m), interpreting
+    the 64 bytes as a little-endian integer, and reducing the resulting integer
+    modulo ``G.Order()``.
```

Comment:
Check if some of those can be matched to RedDSA hashes

Reply:
RedDSA basically uses the same function for HR and H2. I don't think we gain anything by making them them equal in this ZIP since FROST already requires a bunch of domain-separated hash functions anyway.